frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -275,3 +283,77 @@ export function useKubeObjectList<T extends KubeObjectClass>({
     },
   };
 }
+
+/**
+ * Returns a combined list of Kubernetes objects and watches for changes from the clusters given.
+ */
+export function useKubeObjectLists<T extends KubeObjectClass>({
+  kubeObjectClass,
+  namespace,
+  clusters,
+  queryParams,
+}: {
+  /** Class to instantiate the object with */
+  kubeObjectClass: T;
+  /** Object list namespace */
+  namespace?: string;
+  /** Object list clusters */
+  clusters: string[];
+  queryParams?: QueryParameters;
+}): [Array<InstanceType<T>> | null, ApiError | null] &
+  QueryListResponse<KubeList<InstanceType<T>>, InstanceType<T>, ApiError> {
+  const clusterResults = clusters.reduce((acc, cluster) => {
+    acc[cluster] = useKubeObjectList<T>({
```

Comment:
I wonder if we will have problems with this given we are not using the hook in the function body. But we do use some like this already anyway...

Reply:
I wrapped it in useMemo to ensure consistent hook order and such when clusters change.